hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/bytebuddy/EnhancerImpl.java

```diff
@@ -225,6 +225,14 @@ private DynamicType.Builder<?> doEnhance(Supplier<DynamicType.Builder<?>> builde
 					EnhancerConstants.USE_TRACKER_SETTER_NAME
 			);
 
+			builder = addFieldWithGetterAndSetter(
```

Comment:
Might be nice to initialize the field to `-1`

Reply:
We'd need to intercept and enhance all entity constructors to do that. Also, since we don't reset instance ids and we have the `==` check to make sure we have the actual same object, IMO setting this to `-1` doesn't bring much value.

Reply:
AFAIU, you just have to call `.value( -1 )` after `defineField()` e.g.

```
				.defineField( fieldName, type, constants.fieldModifierPRIVATE_TRANSIENT )
						.value( -1 )
						.annotateField( constants.TRANSIENT_ANNOTATION )
```

Reply:
Resetting the value could happen in `org.hibernate.engine.internal.EntityEntryContext#removeEntityEntry` which also manages the entity entry reference.

Reply:
That would result in an `itable stub` invocation, which would degrade performance.

Reply:
Indeed, it seems all constructors would have to initialize this field :/
https://github.com/raphw/byte-buddy/issues/227